src/Emitter.ts

```diff
@@ -1,38 +1,84 @@
 import * as ts from 'typescript';
 import { Dictionary } from './Dictionary';
-import { warn } from './logger';
+import { warn, debug, docletDebugInfo } from './logger';
 import { assertNever } from './assert_never';
-import { isClassDoclet, isNamespaceDoclet, isEnumDoclet } from './doclet_utils';
+import {
+    isDocumentedDoclet,
+    isClassDoclet,
+    isNamespaceDoclet,
+    isEnumDoclet,
+    isDefaultExportDoclet,
+    isNamedExportDoclet,
+    isExportsAssignmentDoclet
+} from './doclet_utils';
 import {
     createClass,
-    createFunction,
+    createClassMember,
     createClassMethod,
-    createInterfaceMethod,
+    createConstructor,
+    createEnum,
+    createFunction,
     createInterface,
-    createClassMember,
     createInterfaceMember,
-    createNamespaceMember,
+    createInterfaceMethod,
     createModule,
+    createExportDefault,
     createNamespace,
+    createNamespaceMember,
     createTypedef,
-    createEnum,
 } from './create_helpers';
+import { generateTree, StringTreeNode, resolveTypeParameters } from './type_resolve_helpers';
 
-interface IDocletTreeNode
+export interface IDocletTreeNode
 {
     doclet: TDoclet;
     children: IDocletTreeNode[];
     isNested?: boolean;
+    /**
+     * Helper flag for named export identification.
+     */
+    isNamedExport?: boolean;
+    /**
+     * Flag set by the 'exported' generation strategy when Emitter._markExported() is called.
+     */
+    isExported?: boolean;
+    /**
+     * Makes the node be exported with another name for export.
+     * Prevails on `@ignore` tags when set.
+     */
+    exportName?: string;
 }
 
-function shouldMoveOutOfClass(doclet: TDoclet)
+function shouldMoveOutOfClass(doclet: TDoclet): boolean
 {
     return isClassDoclet(doclet)
         || isNamespaceDoclet(doclet)
         || isEnumDoclet(doclet)
         || doclet.kind === 'typedef';
 }
 
+function isClassDeclaration(doclet: TDoclet): boolean
+{
+    return !!(
+        doclet && (doclet.kind === 'class')
+        && doclet.meta && (
```

Comment:
Just to explain the reason why I had written this kind of test this way:
The test `doclet.meta` intends to make the further test(s) in the line not throw an exception because `doclet.meta` could be null. Thus, to me, the line had a sense in a whole.
Here the rest of the line is a collection of terms inside the parenthesis, so it is possibly not as clear as other times this pattern is used.

I just wanted to the check this point in case it was not clear enough.
Now, if you feel it better in several lines, that's no problem, please just confirm.

Reply:
I eventually moved this `isClassDeclaration()` function in `doclet_utils.ts` and reformatted the tests as proposed by the way :-)